features/steps/command/shell.feature

```diff
@@ -0,0 +1,101 @@
+Feature: Running shell commands
+
+  You can run an *ad hoc* script with the following step
+  - `When I run the following script:`
+
+  Or you can run shell commands with:
+  - `I run the following (bash|zsh|fish|dash)? commands`
+
+  Background:
+    Given I use a fixture named "cli-app"
+
+  Scenario: Running ruby script
+    Given a file named "features/shell.feature" with:
+    """
+    Feature: Running scripts
+      Scenario: Running ruby script
+        When I run the following script:
+        \"\"\"bash
+        #!/usr/bin/env ruby
+
+        puts "Hello"
+        \"\"\"
+        Then the output should contain exactly "Hello"
+    """
+    When I run `cucumber`
+    Then the features should all pass
+
+  Scenario: Running python script
+    Given a file named "features/shell.feature" with:
+    """
+    Feature: Running scripts
+      Scenario: Running python script
+        When I run the following script:
+        \"\"\"bash
+        #!/usr/bin/env python
+
+        print("Hello")
+        \"\"\"
+        Then the output should contain exactly "Hello"
+    """
+    When I run `cucumber`
+    Then the features should all pass
+
+  Scenario: Running shell commands
+    Given a file named "features/shell.feature" with:
+    """
+    Feature: Running scripts
+      Scenario: Running shell commands
+        When I run the following commands:
+        \"\"\"bash
+        echo "Hello shell"
+        \"\"\"
+        Then the output should contain exactly "Hello shell"
+    """
+    When I run `cucumber`
+    Then the features should all pass
+
+  Scenario: Running bash commands
+    Given a file named "features/shell.feature" with:
+    """
+    Feature: Running scripts
+      Scenario: Running bash commands
+        When I run the following commands with `bash`:
+        \"\"\"bash
+        echo "Hello \c"
+        echo `echo bash` # subshell
+        \"\"\"
+        Then the output should contain exactly "Hello bash"
+    """
+    When I run `cucumber`
+    Then the features should all pass
+
+  Scenario: Running zsh commands
+    Given a file named "features/shell.feature" with:
+    """
+    Feature: Running zsh scripts
+      Scenario: Running zsh commands
+        When I run the following commands in `zsh`:
```

Comment:
It's ok to add another inner scenario for `with` and `in` just to make sure that users directly see the difference in wording.